1_dynamic/future.k

```diff
@@ -16,4 +18,465 @@ endmodule
+                          <task-guards> .Set </task-guards>  // set of guards
+                          <waiting-task-conf> .Bag </waiting-task-conf>
+                        </waiting-task> 
+                      </waiting-tasks>
+                      <active-task>                  // current task
+                        <futureId> 0 </futureId>       // future identifier
+                        <taskState> active </taskState>// state = active/sleeping
+                        <blocked-threads>              // suspended threads
+                          <blocked-thread multiplicity="*">
+                            <guard> .K </guard>          // guard to wake thread
+                            <blocked-thread-conf> .Bag </blocked-thread-conf>
+                          </blocked-thread> 
+                        </blocked-threads>
+                        <threads color="orange">      // running threads
+                           <thread multiplicity="*" color="yellow">
+                              <k color="green"> ($PGM:Stmts ~> execute) </k>
+  
+                              <br/>
+                              <control color="cyan">
+                                <crntClass> Object </crntClass>
+                                <fstack color="blue"> .List </fstack>
+                                <xstack color="purple"> .List </xstack>
+                                <return color="LimeGreen"> void </return>  // KOOL
+                              </control>
+                              <br/>
+                              <env color="violet"> .Map </env>
+                              <holds color="black"> .Map </holds>
+                              <id color="pink"> 0 </id>
+                           </thread>
+                        </threads>
+                        <br/>
+                        <busy color="cyan">.Set </busy>
+                        <terminated color="red"> .Set </terminated>
+                      </active-task>
+                      <nextLoc color="gray"> 0 </nextLoc>
+                      <store color="white"> .Map </store>
+                    </object>
+                  </objects>
+                  <nextObjectId> 1 </nextObjectId>
+                  <nextFutureId> 1 </nextFutureId>
+                  <in color="magenta" stream="stdin"> .List </in>
+                  <out color="brown" stream="stdout"> .List </out>
+                  <br/>
+                  <classes color="Fuchsia">        // KOOL
+                     <class multiplicity="*" color="Fuchsia">
+                        <className color="Fuchsia"> Main </className>
+                        <extends color="Fuchsia"> Object </extends>
+                        <declarations color="Fuchsia"> .K </declarations>
+                     </class>
+                  </classes>
+                </T>
+
+
+/*
+  Object identifiers (references).
+  An object value consists  of a triple (Id, MainClass, CurrentClass), where:
```

Comment:
It seems we have (at least) three values associated with an object:
- object reference (identifier)
- object value
- object state 
We should fix a proper terminology that does not produce confusions.